src/test/java/com/cag/cagbackendapi/integration/UserControllerIntegrationTests.kt

```diff
@@ -139,6 +141,96 @@ class UserControllerIntegrationTests {
     }
 
+    @Test
+    fun updateUser_validInput_200Success() {
+        val headers = HttpHeaders()
+        headers.set("authKey", validAuthKey)
+        val request = HttpEntity(validTestUser, headers)
+
+        val createdUserResponse = testRestTemplate.postForEntity("/user/register", request, String::class.java)
+        val createUser = objectMapper.readValue(createdUserResponse.body, UserResponseDto::class.java)
+        val userId = createUser.user_id
+
+        val validUpdateUser = UserResponseDto(user_id = userId, first_name = "Tony", last_name = "Stark", email="tstark@gmail.com")
+        val headers2 = HttpHeaders()
+        headers2.set("authKey", validAuthKey)
+        val request2 = HttpEntity(validUpdateUser, headers2)
+
+        val updateUserResponse = testRestTemplate.exchange("/user/", HttpMethod.PUT, request2, String::class.java)
+        val updatedUser = objectMapper.readValue(updateUserResponse.body, UserResponseDto::class.java)
+
+        //test created user
+        assertNotNull(createdUserResponse)
+        assertEquals(HttpStatus.CREATED, createdUserResponse.statusCode)
+        assertEquals(validTestUser.first_name, createUser.first_name)
+        assertEquals(validTestUser.email, createUser.email)
+        assertNotNull(createUser.user_id)
+
+        //test updated user
+        assertNotNull(updateUserResponse)
+        assertEquals(HttpStatus.OK, updateUserResponse.statusCode)
+        assertEquals(createUser.user_id, updatedUser.user_id)
+        assertEquals(validUpdateUser.first_name, updatedUser.first_name)
+        assertEquals(validUpdateUser.last_name, updatedUser.last_name)
+        assertEquals(validUpdateUser.email, updatedUser.email)
+        assertNotNull(validUpdateUser.user_id)
+    }
+
+    @Test
+    fun updateUser_missingUserId_400BadRequest() {
+        val invalidUpdateUser = UserResponseDto(null, first_name = "Tony", last_name = "Stark", email="tstark@gmail.com")
+        val headers2 = HttpHeaders()
+        headers2.set("authKey", validAuthKey)
+        val request2 = HttpEntity(invalidUpdateUser, headers2)
+
+        val errorDetailsResponse = testRestTemplate.exchange("/user/", HttpMethod.PUT, request2, ErrorDetails::class.java)
+
+        assertEquals(HttpStatus.BAD_REQUEST, errorDetailsResponse.statusCode)
+        assertNotNull(errorDetailsResponse?.body?.time)
+        assertEquals(errorDetailsResponse?.body?.restErrorMessage, RestErrorMessages.BAD_REQUEST_MESSAGE)
+        assertEquals(errorDetailsResponse?.body?.detailedMessage, DetailedErrorMessages.INVALID_UUID)
+    }
+
+    @Test
+    fun updateUser_invalidAuthKey_401Unauthorized() {
+        val headers = HttpHeaders()
+        headers.set("authKey", validAuthKey)
+        val request = HttpEntity(validTestUser, headers)
+
+        val createdUserResponse = testRestTemplate.postForEntity("/user/register", request, String::class.java)
```

Comment:
You're hitting the register user endpoint in this test instead of the update user one